staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -709,24 +690,27 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 
 	for {
 		startTime := time.Now()
-		getResp, err = s.client.KV.Get(ctx, preparedKey, options...)
+		getResp, err = s.getList(ctx, keyPrefix, opts.Recursive, kubernetes.ListOptions{
+			Revision: withRev,
```

Comment:
Just check - previously we were setting withRev only if it was non-zero.

What is the new semantic when Revision is set to 0? Can you please add a comment here?

Reply:
revision set to 0 is the default. So check for withRev not zero, has no real side effects, just an optimization to avoid allocating another option and appending to list.